tests/fuzz/fuzz-codex32.c

```diff
@@ -0,0 +1,174 @@
+	const char *vec = valid_vectors[idx];
+	size_t len = strlen(vec);
+
+	if (len > max_size)
+		len = max_size;
+
+	memcpy(fuzz_data, vec, len);
+	return len;
+}
+
+void init(int *argc, char ***argv)
+{
+	common_setup("fuzzer");
+}
+
+/* Custom mutator preserving Codex32 structure */
+size_t LLVMFuzzerCustomMutator(uint8_t *fuzz_data, size_t size,
+				size_t max_size, unsigned int seed)
+{
+	srand(seed);
+	char *str = to_string(tmpctx, fuzz_data, size);
+	char *fail;
+	struct codex32 *parts = codex32_decode(tmpctx, NULL, str, &fail);
+
+	/* If invalid, return valid vector */
+	if (!parts)
+		return initial_input(fuzz_data, max_size);
+
+	/* Mutate a random component */
+	switch(rand() % 5) {
+		case 0: /* Mutate HRP */
+			if (streq(parts->hrp, "ms"))
+				parts->hrp = "MS";
+			else
+				parts->hrp = "ms";
+			break;
+		case 1: /* Mutate threshold (0-9) */
+			parts->threshold = rand() % 10;
+			break;
+		case 2: /* Mutate ID (4 characters) */
+			for (int i = 0; i < 4; i++)
+				parts->id[i] = 'A' + (rand() % 26);
+			parts->id[4] = '\0';
+			break;
+		case 3: /* Mutate share index (valid char) */
+			parts->share_idx = "abcdefghijklmnopqrstuvwxyzABCDEF"[rand() % 32];
+			break;
+		case 4: /* Mutate payload */
+			if (tal_bytelen(parts->payload) > 0) {
+			size_t mutate_len = 1 + rand() % tal_bytelen(parts->payload);
+			LLVMFuzzerMutate((u8 *) parts->payload, mutate_len,
+					tal_bytelen(parts->payload));
+			}
+			break;
+	}
+
+	/* Re-encode mutated parts */
+	char *reencoded;
+	const char *err = codex32_secret_encode(tmpctx, parts->hrp, parts->id, parts->threshold,
+				parts->payload, tal_bytelen(parts->payload), &reencoded);
+
+	if (err)
+		return initial_input(fuzz_data, max_size);
+
+	size_t len = strlen(reencoded);
+	if (len > max_size)
+		return initial_input(fuzz_data, max_size);
+
+	memcpy(fuzz_data, reencoded, len);
+	return len;
+}
+
+/* Custom crossover for Codex32 strings */
+size_t LLVMFuzzerCustomCrossOver(const u8 *in1, size_t in1_size,  const u8 *in2, size_t in2_size,
+				u8 *out, size_t max_out_size, unsigned seed)
+{
+	srand(seed);
+	char *str1 = to_string(tmpctx, in1, in1_size);
+	char *str2 = to_string(tmpctx, in2, in2_size);
+	char *fail;
+
+	/* Decode both inputs */
+	struct codex32 *p1 = codex32_decode(tmpctx, NULL, str1, &fail);
+	struct codex32 *p2 = codex32_decode(tmpctx, NULL, str2, &fail);
+
+	/* If either invalid, use initial input */
+	if (!p1 || !p2)
+		return initial_input(out, max_out_size);
+
+	/* Choose component to crossover */
+	struct codex32 *child = p1;
+	switch(rand() % 5) {
+		case 0:
+			child->hrp = p2->hrp;
+			break;
+		case 1:
+			child->threshold = p2->threshold;
+			break;
+		case 2:
+			memcpy(child->id, p2->id, 5);
+			break;
+		case 3:
+			child->share_idx = p2->share_idx;
+			break;
+		case 4: /* Payload crossover */
+			child->payload = tal_arr(child, u8, tal_bytelen(p2->payload));
+			memcpy((u8 *) child->payload, p2->payload, tal_bytelen(p2->payload));
+			break;
```

Comment:
The idea behind the crossover mutator is to combine pieces of each input into a new input (like genetic crossover).  Using the `cross_over` helper function from `libfuzz.h` here would greatly strengthen this mutator.